### PR TITLE
LINK-643, LINK-644 | Edit registration page and registration list page

### PR DESCRIPTION
**Pull request by jorilindell**
## Description
Integrating registration list page with LE API.
Sending registration search params to BE. BE is not handling them yet. 
Deleting a single registration from registration list view.
Integrating edit registration page with LE API
Note that there is typo in BE side and because of that the enrolment start and end times are not saved. It will be fixed in https://helsinkisolutionoffice.atlassian.net/browse/LINK-662
Deleting a single registration from edit registration page

## Closes 
[LINK-643](https://helsinkisolutionoffice.atlassian.net/browse/LINK-643)
[LINK-644](https://helsinkisolutionoffice.atlassian.net/browse/LINK-644)

## Screenshots
Screenshot 2021-11-04 at 19 30 31

Screenshot 2021-11-04 at 19 30 42

Screenshot 2021-11-04 at 19 30 48

Screenshot 2021-11-04 at 19 32 21

Screenshot 2021-11-04 at 19 32 30

## Testing 
Can be tested at http://localhost:3000/fi/registrations

### CODE REVIEW

**Review by Sartonon — COMMENTED** [review]
Looks good. Couple minor comments added.

Also something strange happening with html attributes :)
Screenshot 2021-11-15 at 9 31 40

File: src/domain/registration/EditRegistrationPage.tsx [hunk 1]
```diff
@@ -25,13 +38,21 @@ import ConfirmationMessageSection from './formSections/confirmationMessageSectio
...
-import { getRegistrationInitialValues } from './utils';
+import { getRegistrationInitialValues, registrationPathBuilder } from './utils';
 import { registrationSchema, showErrors } from './validation';
 
 interface EditRegistrationPageProps {
+  refetch: (
+    variables?: Partial<RegistrationQueryVariables>
+  ) => Promise<ApolloQueryResult<RegistrationQuery>>;
   registration: Registration;
 }
```
[the remark this answers is not in the document]
Comment (jorilindell) [reply src/domain/registration/EditRegistrationPage.tsx:58]: Renamed

File: src/domain/registration/EditRegistrationPage.tsx [hunk 2]
```diff
@@ -48,8 +84,35 @@ const EditEventPage: React.FC<EditRegistrationPageProps> = ({
...
+    deleteRegistration({
+      onSuccess: () => goToRegistrationsPage(),
+    });
+  };
+
+  const onUpdate = (values: RegistrationFormFields) => {
+    updateRegistration(values, {
+      onError: (error: any) => showServerErrors({ error }),
+      onSuccess: async () => {
+        // await refetch();
```
[the remark this answers is not in the document]
Comment (jorilindell) [reply src/domain/registration/EditRegistrationPage.tsx]: Oh, forgot to uncomment this. Uncommented

File: src/domain/registration/hooks/useRegistrationUpdateActions.ts [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+}: Props): UseEventUpdateActionsState => {
+  const isMounted = useIsMounted();
+  const locale = useLocale();
+  const apolloClient = useApolloClient() as ApolloClient<NormalizedCacheObject>;
+  const { user } = useUser();
+  const location = useLocation();
+  const [saving, setSaving] = React.useState<REGISTRATION_EDIT_ACTIONS | false>(
+    false
+  );
+  const [openModal, setOpenModal] = React.useState<MODALS | null>(null);
```
[the remark this answers is not in the document]
Comment (jorilindell) [reply src/domain/registration/hooks/useRegistrationUpdateActions.ts]: Implemented

**Pull request merged**